src/settings/environment.rs

```diff
@@ -0,0 +1,109 @@
+use std::collections::HashMap;
+use std::env;
+
+use config::{Source, Value};
+
+pub trait QueryEnvironment {
+    fn get_var(&self, var: &'static str) -> Result<String, std::env::VarError>;
+
+    fn empty(&self) -> Result<bool, failure::Error>;
```

Comment:
this is a convenience method. i'm not... 100% certain about keeping this here, but yeah it does "query" the environment.